ts/defs.bzl

```diff
@@ -136,7 +136,21 @@ def ts_project(
         data: Files needed at runtime by binaries or tests that transitively depend on this target.
             See https://bazel.build/reference/be/common-definitions#typical-attributes
 
-        deps: List of labels of other rules that produce TypeScript typings (.d.ts files)
+        deps: List of labels of other rules that produce TypeScript typings (`.d.ts` files)
+
+            If this list contains linked npm packages, npm package store targets or other targets that provide
+            `JsInfo`, `NpmPackageStoreInfo` providers are gathered from `JsInfo`. This is done directly from
+            the `npm_package_store_deps` field of these. For linked npm package targets, the underlying
+            `npm_package_store` target(s) that back the links is used. Gathered `NpmPackageStoreInfo`
+            providers are propagated to the direct dependencies of downstream linked `npm_package` targets.
```

Comment:
I think "direct dependencies" is confusing here, since we're talking about indirect deps aren't we? (even if it's implemented by propagating to the direct deps).

Can we say something such as "providers are available to downstream linked `npm_package` targets"?

Reply:
Hmm I think "direct dependencies of downstream linked packages" is correct. Basically saying that they go in the `dependencies` of the downstream linked package if you were to translate to `package.json` terms

Reply:
Yeah I think it's correct, just confusing with so many keywords in there. I had to read it multiple times :grimacing: